### ISSUE TITLE
checkbox: sizing does not match spec or accessibility guidance

### ISSUE THREAD

**Issue by aipsensmall** [angular/material#9927]
**Actual Behavior**:
-  `What is the issue? *` Checkboxes (selected and not) are 20px squares. They are also not within a 24px bounding box.
-  `What is the expected behavior?` As per [Material icon library](https://material.io/icons/#ic_check_box_outline_blank), checkboxes should be 18px squares within a 24px by 24px square container.

@ErinCoughlan

**Comment by clshortfuse** [comment]
This PR has been on the backburner, but it would close this issue: https://github.com/angular/material/pull/8713
